packages/block-library/src/navigation/style.scss

```diff
@@ -494,147 +487,104 @@ button.wp-block-navigation-item__content {
 		color: inherit !important;
 		background-color: inherit !important;
 	}
+}
 
-	// Overlay menu.
-	// Provide an opinionated default style for menu items inside.
-	// Inherit as much as we can regarding colors, fonts, sizes,
-	// but otherwise provide a baseline.
-	// In a future version, we can explore more customizability.
-	&.is-menu-open {
-		display: flex; // Needs to be set to override "none".
-		flex-direction: column;
-		background-color: inherit;
-
-		// Animation.
-		animation: overlay-menu__fade-in-animation 0.1s ease-out;
-		animation-fill-mode: forwards;
-		@include reduce-motion("animation");
-
-		// Try to inherit any root paddings set, so the X can align to a top-right aligned menu.
-		padding-top: clamp(1rem, var(--wp--style--root--padding-top), 20rem);
-		padding-right: clamp(1rem, var(--wp--style--root--padding-right), 20rem);
-		padding-bottom: clamp(1rem, var(--wp--style--root--padding-bottom), 20rem);
-		padding-left: clamp(1rem, var(--wp--style--root--padding-left), 20em);
-
-		// Allow modal to scroll.
-		overflow: auto;
-
-		// Give it a z-index just higher than the adminbar.
-		z-index: 100000;
+// Overlay menu.
+// Provide an opinionated default style for menu items inside.
+// Inherit as much as we can regarding colors, fonts, sizes,
+// but otherwise provide a baseline.
+// In a future version, we can explore more customizability.
+.wp-block-navigation__responsive-container.is-menu-open {
```

Comment:
Why don't we move this to the other file too?

Reply:
If we want to make an agnostic component, Dave's idea is correct, but we do want to separate things that are specific to the navigation if they pertain to the wider viewport or to the collapsed version of it. Where that should live is another matter.

Reply:
I was thinking about this some more. This type of overlay may well be specific to the Navigation block. I still think it's helpful to extract the overlay styles from other behaviour that handles sub blocks and things.

This code is unlikely to end up being a standalone `<Overlay>` component because it's heavily tied to a specific responsive behaviour which shows and hides specific pieces of the overlay depending on an arbitrary breakpoint. That is unlikely to be portable to other scenarios.

Reply:
I think you might be right. We can refactor again if we feel like we need to, and it will already be way cleaner than it was anyway